examples/src/pages/Migration.tsx

```diff
@@ -19,6 +19,7 @@ export function Migration() {
   const canvasWrapperRef = useRef<HTMLDivElement>(null);
   useEffect(() => {
     const gui = new dat.GUI();
+    let viewer: Cognite3DViewer;
```

Comment:
Probably part of another pr, but thought I would ask since you are making the viewer a let variable.
Should we not rather do something like
```
cons [setClient, client] = useState();

useEffect(() => {
  const viewer = new Cognite3DViewer();
  return () => {
    viewer.dispose();
  }
}, [client])
```

Or something along those lines?

Reply:
@j-bjorne We probably should. At least for such things as the viewer (where it makes sense). 

Here it makes sense to have `let` because we should clean up things when component is unmounted. But I don't see the need to declare it as a state, because nothing else needs it. 

We also need to clean up after the reveal manager in a similar way. That's why I asked about `dispose()` for the reveal manager yesterday. 

About putting `client` as a state thing, it doesn't really make sense to me since it's declared and used inside the same effect and doesn't change. But it would be exactly as you wrote if to lift client declaration higher (as a context or prop, for example). 

I promised to make at least some examples to be more in "react way" and hopefully will deal with it once I have some spare time in that sprint.